homeassistant/components/google_assistant/trait.py

```diff
@@ -873,6 +892,103 @@ def query_attributes(self):
         )
 
 
+@register_trait
+class ArmDisArmTrait(_Trait):
+    """Trait to Arm or Disarm a Security System.
+
+    https://developers.google.com/actions/smarthome/traits/armdisarm
+    """
+
+    name = TRAIT_ARMDISARM
+    commands = [COMMAND_ARMDISARM]
+
+    state_to_service = {
+        STATE_ALARM_ARMED_HOME: SERVICE_ALARM_ARM_HOME,
+        STATE_ALARM_ARMED_AWAY: SERVICE_ALARM_ARM_AWAY,
+        STATE_ALARM_ARMED_NIGHT: SERVICE_ALARM_ARM_NIGHT,
+        STATE_ALARM_ARMED_CUSTOM_BYPASS: SERVICE_ALARM_ARM_CUSTOM_BYPASS,
+        STATE_ALARM_TRIGGERED: SERVICE_ALARM_TRIGGER,
+    }
+
+    @staticmethod
+    def supported(domain, features, device_class):
+        """Test if state is supported."""
+        return domain == alarm_control_panel.DOMAIN
+
+    @staticmethod
+    def might_2fa(domain, features, device_class):
+        """Return if the trait might ask for 2FA."""
+        return True
+
+    def sync_attributes(self):
+        """Return ArmDisarm attributes for a sync request."""
+        response = {}
+        levels = []
+        for state in self.state_to_service:
+            level = {
+                "level_name": state,
+                "level_values": [
+                    {
+                        "level_synonym": [
+                            state.replace("_", " "),
+                            state.split("_")[
```

Comment:
What's going on here ? why?

Reply:
Please add a comment

Reply:
I wanted to generate the synonyms from the state name itself. so `armed_away` turns into `armed away` or `away` instead of hard coding it.
Do you recommend to keep it this way or just had code the synonyms?